pthread_util.c

```diff
@@ -39,14 +54,30 @@ int create_thread(pthread_t *thread,  void *(*start_routine) (void *), void *arg
 	if (detached) {
 		pthread_attr_init(&attr);
 		pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_DETACHED);
+/* On MacOS X we can only set a threads name from inside that thread.
+Because of this, if we're compiling for OS X, we have pthread_create
+specify a trampoline function that calls the actual routine that the 
+thread will be running. */
+#ifndef __APPLE__
 		rc = pthread_create(thread, &attr, start_routine, arg);
+#else
+		//FIXME: This could be a potential memory leak. Not  sure if
+		//we can free t_args immediately without messing up threading.
+		t_args = malloc(sizeof(struct pthread_osx_trampoline_args));
+		t_args->thread_name = name;
+		t_args->thread_start = start_routine;
+		t_args->thread_args = arg;
+		rc = pthread_create(thread, &attr, pthread_osx_trampoline, (void *) t_args);
+#endif
 		pthread_attr_destroy(&attr);
 	} else {
 		rc = pthread_create(thread, NULL, start_routine, arg);
 	}
 #ifdef _GNU_SOURCE
+#ifndef __APPLE__
```

Comment:
So presumably on linux, a thread can set its own name.  I wonder if it would be better to make the linux code work more like the apple code, and use the trampoline in all cases, and just call pthread_setname_np() with one or two arguments depending on the ifdef in the trampoline.  Would doing that reduce the amount of code under ifdef to just the call to pthread_setname_np()?

Reply:
That's what I was thinking as well. My goal initially was to impact existing code as little as possible, so I left that out for the time being. Maybe the pthread handling code should all be consolidated into pthread_utils.c? I'm not sure it's used everywhere.

Reply:
> Maybe the pthread handling code should all be consolidated into pthread_utils.c? I'm not sure it's used everywhere.

(Without looking at the code, he says) I think the ssgl stuff probably doesn't use it, because at one time I imagined ssgl might stand alone as a generic lobby independent of the rest (though I now think this is not very realistic to imagine.)  I think everywhere else in SNIS the pthread_utils is used for thread creation.

Reply:
For convenience, I can take this as is, and modify it later. No need to hold up good functionality for code aesthetics that can easily be fixed later.

Reply:
Ahh yeah, that makes sense. I did see those usages mostly in the ssgl code. Sounds good to me. My original intention was to develop a baseline that could be used elsewhere :D